File: Examples/CSharp/Loading-and-Saving/ExtractRaw3DContentsFromPdf.cs
using System;
using System.IO;
using System.Text;
using System.Collections.Generic;
using Aspose.ThreeD;
using Aspose.ThreeD.Formats;

namespace Aspose._3D.Examples.CSharp.Loading_Saving
{
    class ExtractRaw3DContentsFromPdf
    {
        public static void Run()
        {
            //ExStart:ExtractRaw3DContentsFromPdf
            // The path to the documents directory.
            string MyDir = RunExamples.GetDataDir();
            byte[] password = null;
            // Extract 3D contents
            List<byte[]> contents = FileFormat.PDF.Extract(MyDir + "House_Design.pdf", password);
            int i = 1;
            // Iterate through the contents and in separate 3D files
            foreach (byte[] content in contents)
            {
                string fileName = "3d-" + (i++);
                File.WriteAllBytes(fileName, content);
            }
            //ExEnd:ExtractRaw3DContentsFromPdf            
        }
    }
}

File: Examples/VisualBasic/Loading-and-Saving/ExtractRaw3DContentsFromPdf.vb
Imports System.IO
Imports System.Text
Imports System.Collections.Generic
Imports Aspose.ThreeD
Imports Aspose.ThreeD.Formats

Namespace Loading_Saving
    Class ExtractRaw3DContentsFromPdf
        Public Shared Sub Run()
            ' ExStart:ExtractRaw3DContentsFromPdf
            ' The path to the documents directory.
            Dim MyDir As String = RunExamples.GetDataDir()
            Dim password As Byte() = Nothing
            ' Extract 3D contents
            Dim contents As List(Of Byte()) = FileFormat.PDF.Extract(MyDir & Convert.ToString("House_Design.pdf"), password)
            Dim i As Integer = 1
            ' Iterate through the contents and in separate 3D files
            For Each content As Byte() In contents
                Dim fileName As String = "3d-" + (System.Math.Max(System.Threading.Interlocked.Increment(i), i - 1))
                File.WriteAllBytes(fileName, content)
            Next
            ' ExEnd:ExtractRaw3DContentsFromPdf            
        End Sub
    End Class
End Namespace
